### PR TITLE
[Workflow] MWPW-134543 Dispatch DC workflow when main is changed

### PR DESCRIPTION
**Pull request by TsayAdobe**
* It creates a workflow dispatch to DC repo when there is a change to main
* DC repo will trigger necessary tests to ensure the change in Milo doesn't break DC pages
* Will add PRs later when shift-left is needed
* Other consuming teams can add their dispatches to jobs

Resolves: [MWPW-134543](https://jira.corp.adobe.com/browse/MWPW-134543)

**Test URLs:**
- N/A

### CODE REVIEW

**Review by mokimo — APPROVED** [review]
Cool, do you think other consumers would benefit from this as well?

File: .github/workflows/dispatch.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+      - main
+
+jobs:
+  dispatch-dc:
+    name: Dispatch DC workflow
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out repository
+        uses: actions/checkout@v3    
+      - uses: dorny/paths-filter@v2
```
Comment (JackySun9) [inline .github/workflows/dispatch.yml:15]: do you have v3 for this

File: .github/workflows/dispatch.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+          filters: |
+            src:
+              - 'libs/**'    
+      - if: steps.changes.outputs.src == 'true'
+        name: Trigger DC Workflow
+        uses: actions/github-script@v6
+        with:
+          github-token: ${{ secrets.DC_PAT }}
+          script: |
+            github.rest.actions.createWorkflowDispatch({
```
Comment (JackySun9) [inline .github/workflows/dispatch.yml:27]: do you need error handling for this

File: .github/workflows/dispatch.yml (hunk 1 above)
Comment (TsayAdobe) [reply .github/workflows/dispatch.yml:15]: There is only v2.
https://github.com/dorny/paths-filter

File: .github/workflows/dispatch.yml (hunk 2 above)
Comment (TsayAdobe) [reply .github/workflows/dispatch.yml:27]: The action library actions/github-script@v6 should have handled most errors. Not sure what is not handled. We will handle it when we see it.

**Comment by TsayAdobe** [comment]
> Cool, do you think other consumers would benefit from this as well?

Yes, if they want to do some actions or tests in their repo when Milo is changed. They can add a workflow dispatch in this script.

**Review by JackySun9 — APPROVED** [review]

**Pull request merged**